src/Domain/Donation.php

```diff
@@ -95,16 +93,16 @@ class Donation extends SalesforceWriteProxy
      * @ORM\Column(type="string", length=3)
      * @var string  ISO 4217 code for the currency in which all monetary values are denominated, e.g. 'GBP'.
      */
-    protected string $currencyCode;
+    protected readonly string $currencyCode;
```

Comment:
Doctrine says "An entity class must not be final nor read-only but it may contain final methods or read-only properties."